### PR TITLE
feat(alerts): Alerts details redesign route

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds a new alert rule details route behind the redesign flag. It copies the elements of the incident details page but they will be modified soon.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/rules/index.tsx [hunk 1]
```diff
@@ -104,9 +104,12 @@ class AlertRulesList extends AsyncComponent<Props, State & AsyncComponent['state
   }
 
   renderList() {
+    const {
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/rules/index.tsx:107]: pretty cool destructure 🏅

File: src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+          </Controls>
+        </BreadCrumbBar>
+        <Details columns={isErrorDataset ? 5 : 3}>
+          <div>
+            <IncidentTitle data-test-id="incident-title" loading={!isIncidentReady}>
+              {incident && !hasIncidentDetailsError ? incident.title : 'Loading'}
+            </IncidentTitle>
+            <IncidentSubTitle loading={!isIncidentReady}>
+              {t('Triggered: ')}
+              {dateStarted}
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx:120]: can we use `<DateTime>` here

File: src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      incident,
+      params,
+      stats,
+      onSubscriptionChange,
+    } = this.props;
+    const isIncidentReady = !!incident && !hasIncidentDetailsError;
+    // ex - Wed, May 27, 2020 11:09 AM
+    const dateFormat = use24Hours() ? 'ddd, MMM D, YYYY HH:mm' : 'llll';
+    const dateStarted =
+      incident && moment(new Date(incident.dateStarted)).format(dateFormat);
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx:82]: if we're using datetime to format we can reuse dateStarted moment object to get the duration

**Review by davidenwang — APPROVED** [review]
lgtm, i understand we'll be overhauling the header/body files with code for the new layout

File: src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx (hunk 2 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx:120]: 👍

File: src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx (hunk 3 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx]: I had to do it a bit differently but these will likely change in the next pr in this epic.

**Pull request merged**